server/s3_get_bucket_action.cc

```diff
@@ -322,6 +322,11 @@ void S3GetBucketAction::get_next_objects_successful() {
       }
     }
 
+    if (kv.first.rfind(SYM_EXT_OBJ_PREFIX, 0) != std::string::npos) {
```

Comment:
since ~XXX will occur at end, hence its better if you first check outside the loop initially whether the first character of last record's key is ~ or { or } or |  if so then set a flag and here within the loop if that flag is set then check for the first character of the key to be ~
In that way we can avoid the check for each key

Also rfind searches for the last. we need to see the first character of the key to be ~

Reply:
> EXTENDED_METADATA_OBJECT_PREFIX ?

ok. will use it

Reply:
> since ~XXX will occur at end, hence its better if you first check outside the loop initially whether the first character of last record's key is ~ or { or } or | if so then set a flag and here within the loop if that flag is set then check for the first character of the key to be ~
> In that way we can avoid the check for each key
> 
> Also rfind searches for the last. we need to see the first character of the key to be ~

rfind(), when passed with 0 as index, tries to find search string at 0th index position and less. So, it tries to search at start of string.

Will do changes as per suggestion by "Rajesh N" to avoid searching for "~" in each result set. For this, search each key for "~" in only result set that has "~" (this will be the last result set).